src/MLIAP/mliap_descriptor_so3.cpp

```diff
@@ -0,0 +1,298 @@
+/* ----------------------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   https://lammps.sandia.gov/, Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+#include "mliap_descriptor_so3.h"
+#include "atom.h"
+#include "comm.h"
+#include "error.h"
+#include "memory.h"
+#include "mliap_data.h"
+#include "pair_mliap.h"
+#include "mliap_so3.h"
+
+#include <cmath>
+#include <cstring>
+
+using namespace LAMMPS_NS;
+
+#define MAXLINE 1024
+#define MAXWORD 3
+
+
+MLIAPDescriptorSO3::MLIAPDescriptorSO3(LAMMPS *lmp,
+                                       char *paramfilename):
+  MLIAPDescriptor(lmp)
+{
+  nelements = 0;
+  elements = nullptr;
+  radelem = nullptr;
+  wjelem = nullptr;
+  so3ptr = nullptr;
+  read_paramfile(paramfilename);
+
+  so3ptr = new MLIAP_SO3(lmp,rcutfac,lmax,nmax,alpha);
+
+  ndescriptors = so3ptr->ncoeff;
+}
+
+MLIAPDescriptorSO3::~MLIAPDescriptorSO3()
+{
+  delete so3ptr;
```

Comment:
This looks as if there would be memory leakage due to missing `delete` statements. Have you checked whether your code is "valgrind clean"?

Reply:
I have checked memory leak with "valgrind". I found no leak adding those lines.